backend/app/usecase/usecase_test.go

```diff
@@ -0,0 +1,195 @@
+	}{
+		{
+			name: "user already signed in",
+			now:  now,
+			githubIDProvider: stubIDProvider{
+				authorizationURL: "github_sign_in_link",
+				accessToken:      "access_token",
+			},
+			authToken:                    "auth_token",
+			tokenValidDuration:           time.Hour,
+			expectedShowHomeCallArgs:     []showHomeCallArgs{},
+			expectedShowUserHomeCallArgs: []showUserHomeCallArgs{},
+			expectedShowExternalPageCallArgs: []showExternalPageCallArgs{
+				{
+					link: "github_sign_in_link",
+				},
+			},
+		},
+		{
+			name: "user has not signed in",
+			now:  now,
+			githubIDProvider: stubIDProvider{
+				authorizationURL: "github_sign_in_link",
+				accessToken:      "access_token",
+			},
+			authToken:                    "",
+			tokenValidDuration:           time.Hour,
+			expectedShowUserHomeCallArgs: []showUserHomeCallArgs{},
+			expectedShowExternalPageCallArgs: []showExternalPageCallArgs{
+				{
+					link: "github_sign_in_link",
+				},
+			},
+		},
+		{
+			name: "auth token has no email",
+			now:  now,
+			githubIDProvider: stubIDProvider{
+				authorizationURL: "github_sign_in_link",
+				accessToken:      "access_token",
+			},
+			authToken: `
+{
+  "issued_at": "2020-01-26T08:32:40.759788656Z"
+}
+`,
+			tokenValidDuration:           time.Hour,
+			expectedShowUserHomeCallArgs: []showUserHomeCallArgs{},
+			expectedShowExternalPageCallArgs: []showExternalPageCallArgs{
+				{
+					link: "github_sign_in_link",
+				},
+			},
+		},
+		{
+			name: "auth token has empty email",
+			now:  now,
+			githubIDProvider: stubIDProvider{
+				authorizationURL: "github_sign_in_link",
+				accessToken:      "access_token",
+			},
+			authToken: `
+{
+  "email": "",
+  "issued_at": "2020-01-26T08:32:40.759788656Z"
+}
+`,
+			tokenValidDuration:           time.Hour,
+			expectedShowUserHomeCallArgs: []showUserHomeCallArgs{},
+			expectedShowExternalPageCallArgs: []showExternalPageCallArgs{
+				{
+					link: "github_sign_in_link",
+				},
+			},
+		},
+		{
+			name: "auth token expired",
+			now:  now,
+			githubIDProvider: stubIDProvider{
+				authorizationURL: "github_sign_in_link",
+				accessToken:      "access_token",
+			},
+			authToken: `
+{
+  "email": "byliuyang11@gmail.com",
+  "issued_at": "2020-01-26T07:31:40.759788656Z"
+}
+`,
+			tokenValidDuration:           time.Hour,
+			expectedShowUserHomeCallArgs: []showUserHomeCallArgs{},
+			expectedShowExternalPageCallArgs: []showExternalPageCallArgs{
+				{
+					link: "github_sign_in_link",
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			useCase := newUseCase(
+				testCase.now,
+				testCase.existingURLs,
+				testCase.tokenValidDuration,
+				testCase.githubIDProvider,
+			)
+			presenter := newMockPresenter()
+			useCase.RequestGithubSignIn(testCase.authToken, &presenter)
+
+			mdtest.Equal(t, testCase.expectedShowUserHomeCallArgs, presenter.showUserHomeCallArgs)
```

Comment:
We should check `expectedShowHomeCallArgs` value

Reply:
Done.